src/Data/Array/ST.js

```diff
@@ -28,6 +28,16 @@ exports.poke = function (i) {
   };
 };
 
+exports.popImpl = function (just) {
+  return function (nothing) {
+    return function (xs) {
+      return function () {
+        return xs.length ? just(xs.pop()) : nothing;
```

Comment:
I'm not particularly keen on relying on the falsiness of 0. Can you use `xs.length > 0` for the condition here instead please?